src/components/modal/ModalDialog.js

```diff
@@ -112,12 +112,14 @@ class ModalDialog extends React.Component<Props> {
 
         return (
             <div ref={modalRef} className={classNames('modal-dialog', className)} {...divProps}>
-                <div className="modal-header">
-                    <h2 className="modal-title" id={`${this.modalID}-label`}>
-                        {title}
-                    </h2>
+                <div className="modal-header-container">
+                    <div className="modal-header">
+                        <h2 className="modal-title" id={`${this.modalID}-label`}>
+                            {title}
+                        </h2>
+                    </div>
+                    {this.renderCloseButton()}
```

Comment:
Since the close button has `absolute` positioning, it seems odd to put this inside of an element that also has `fixed` / `absolute` positioning since the two elements would not have an effect on each other.

**However**, if the purpose of moving this close button inside the `modal-header-container` is to vertically align the button with the title, then this change does make sense.

In that case, I think it would be better to change [this line](https://github.com/box/box-ui-elements/blob/173fb3328488e2001f99ef16fcaa0ca001459025/src/components/modal/Modal.scss#L175) to `height: 100%;`.

Reply:
This was the intent, to align the close button and the title. I've updated it to `height: 100%`.